src/handbook/development/ops/maintenance.md

```diff
@@ -0,0 +1,76 @@
+---
+navTitle: Infrastructure Maintenance
+---
+
+Periodically, FlowFuse performs maintenance on FlowFuse Cloud resources. This page describes how we manage and execute maintenance tasks around the infrastructure.
+
+# Maintenance scope
+
+The maintenance most often involves updates to the following resources:
+* underlying cluster hosts
+* Kubernetes versions
+* databases
+* networking components
+
+ 
+# Maintenance process
+
+## Planning
+
+The maintenance process is initiated by the Operations Team. The process is planned in advance and communicated to the customers. The maintenance window is scheduled to minimize the impact on the team and customers.
+
+Within a team, a maintenance date should be agreed upon, and the team should be informed about the maintenance window.
+
+While picking the correct date and time for maintenance, the following should be considered:
+* at least two weeks' notice should be provided to customers
+* maintenance should not overlap with [scheduled release](../releases/process.md) week
+
+## Customers communication
+
+Once the maintenance date is set, customers should be informed about the upcoming maintenance. The communication should include:
+* the scope of the maintenance
+* the date and time of the maintenance
+* expected duration of the maintenance
+* self-care actions which customers can/should take before the maintenance date
```

Comment:
How can they prevent this? The way **our** infra is set up influences all this, and we never get this from Heroku et al. Why do we get this from FlowFuse?

Reply:
It's fundamental to how kubernetes works.

You can only have a version mismatch of +- 2 (e.g. 1.24 to 1.26) between the core kubernetes api and the version of the kubelet managing the node the pods are running on.

So to upgrade kubernetes version you upgrade the core api, then you need to create new nodes, and then migrate the running pods to those new nodes. The migration requires stopping the instance on the old node and starting it on the new version.

In theory we could start the instance, but that does end up with 2 instances running for a short while, which requires planning in the flow design (around caching or state management)

Reply:
@hardillb My point is that other compute providers just restart, even cloud providers. Somehow we don't and send emails regularly to customers asking them to restart instances, and we don't even tell them which ones.